### ISSUE TITLE
Can't access downloaded files when using remote browser

### ISSUE THREAD

**Issue by Estavaringo** [go-rod/rod#772]
Rod Version: v0.112.0

Hi! I'm facing an issue when trying to access downloaded files of a remotely managed browser.

## The code to demonstrate your question

Before running the code you need to run the rod image  `docker run -p 7317:7317 ghcr.io/go-rod/rod`

see: https://go-rod.github.io/#/custom-launch?id=remotely-manage-the-launcher

```go
func TestRod(t *testing.T) {
	l := launcher.MustNewManaged("")

	browser := rod.New().
		Client(l.MustClient()).
		MustConnect()
	defer browser.MustClose()

	page := browser.MustPage("https://file-examples.com/index.php/sample-documents-download/sample-pdf-download/")

	wait := browser.MustWaitDownload()

	page.MustElementR("a", "DOWNLOAD SAMPLE PDF FILE").MustClick()

	_ = utils.OutputFile("t.pdf", wait()) //test fail here trying to access downloaded file
}

```

## What you got

➜ go test -run TestRod
parallel test 8
--- FAIL: TestRod (6.23s)
panic: open /tmp/rod/downloads/980336a8-57d8-4bb2-be56-50d0cdca9670: no such file or directory [recovered]

## What you expected to see

➜ go test -run TestRod
parallel test 8
PASS
ok      github.com/go-rod/rod   6.615s

Test passed and file t.pdf created

## What have you tried to solve the question

Using a local browser works fine and the file is created.

I think (maybe) that I can make this work by [hijacking](https://github.com/go-rod/rod/blob/master/examples_test.go#L463) the requests and getting the response from the hijack function, but I'm not quite sure because inspecting the browser network resources doesn't show any http request with the downloaded file in the response.

Is there any way to make the `MustWaitDownload()` function work with a remote browser?

**Comment by ysmood** [comment]
A naive solution:

```go
p := "/tmp/myfile"

wait := browser.WaitDownload(p)()

page.MustElementR("a", "DOWNLOAD SAMPLE PDF FILE").MustClick()

GetFileByHTTP(p) // implement it yourself
```

Use the `ghcr.io/go-rod/rod` as a base image and create a static file server on it, the server should serve the `/tmp` directory.

Run the rod-manager and your static file server on the container at the same time with different port.

**Comment by Estavaringo** [comment]
Thanks for the answer, it worked as expected.

I was already running a custom manager server, with my own built image, so I just had to start a file server in a different port, as you said.

The working code looks like this:

``` go
func TestRod(t *testing.T) {
	l := launcher.MustNewManaged("")

	browser := rod.New().
		Client(l.MustClient()).
		MustConnect()
	defer browser.MustClose()

	page := browser.MustPage("https://file-examples.com/index.php/sample-documents-download/sample-pdf-download/")

	dirName := "test"
	wait := browser.WaitDownload(filepath.Join(os.TempDir(), dirName))

	page.MustElementR("a", "DOWNLOAD SAMPLE PDF FILE").MustClick()

	info := wait()

	resp, err := http.Get(fmt.Sprintf("http://localhost:8000/%s", filepath.Join(dirName, info.GUID)))
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()
	body, err := io.ReadAll(resp.Body)
	if err != nil {
		panic(err)
	}
	_ = utils.OutputFile("t.pdf", body)
}

```

This approach has a problem that the files are not automatically deleted after being read by the client. So I changed to use the same dir as the user data dir for saving the files, because the user data dir is always deleted when the browser is closed.

Thanks for your help! I think this is an amazing workaround for this problem.

**Comment by ysmood** [comment]
I will add another endpoint for the static file server for file deletions.

**Comment by Estavaringo** [comment]
Hey @ysmood 
I included another endpoint on the static file server for file deletions and it worked nice.

But I have another problem now. 
I'm running the browsers in a kubernetes cluster with an autoscale enabled, alongside with the file server.

The problem is that sometimes my client tries to get the download file from a pod other than the one running the browser, causing a 404 file not found.

Do you know how to use a static file server in a distributed architecture like this?

**Comment by ysmood** [comment]
> Is there any chance to get the downloaded file using the websocket that communicate with the browser?

No, cdp doesn't allow you to do that yet, I also hope they can add this api.
